ihmc-communication/src/main/java/us/ihmc/communication/ROS2Tools.java

```diff
@@ -89,7 +96,18 @@ public static ROS2Node createROS2Node(PubSubImplementation pubSubImplementation,
     */
    public static ROS2Node createLoopbackROS2Node(PubSubImplementation pubSubImplementation, String nodeName)
    {
-      return new ROS2Node(pubSubImplementation, nodeName, FACTORY.getDomainId(), InetAddress.getLoopbackAddress());
+      InetAddress loopbackAddress = InetAddress.getLoopbackAddress();
+      if (FACTORY.getAddressRestriction().length != 0 && !Arrays.asList(FACTORY.getAddressRestriction()).contains(loopbackAddress))
+      {
+         LogTools.warn("""
+                             You are creating a loopback ROS2 node, but your RTPSSubnet restriction does not allow the loopback address.
+                             For this node to communicate with other nodes, you can add {}/8 to the network parameters file.
+                             (Currently allowed addresses: {})""",
```

Comment:
Minor: Formatting and indent is Kinda weird

Reply:
Yeah, I agree. That's what the auto formatter gave me. I'll change it.